### PR TITLE
PRO-2834 email notifications

### PR DESCRIPTION
**Pull request by malinovsky**

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/probate/config/notifications/NotificationTemplates.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.validation.annotation.Validated;
+import uk.gov.hmcts.probate.model.ApplicationType;
+
+import javax.validation.Valid;
+import java.util.Map;
+
+@Getter
+@Setter
+@Component
+@Validated
```
Comment (sanjayparekh-hmcts) [inline src/main/java/uk/gov/hmcts/probate/config/notifications/NotificationTemplates.java]: Does this class need to be @Validated?

File: src/main/java/uk/gov/hmcts/probate/config/notifications/NotificationsProperties.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.annotation.Validated;
+
+import javax.validation.Valid;
+
+@Getter
+@Setter
+@Component
+@Validated
```
Comment (sanjayparekh-hmcts) [inline src/main/java/uk/gov/hmcts/probate/config/notifications/NotificationsProperties.java]: same here

File: src/main/java/uk/gov/hmcts/probate/config/properties/registries/RegistriesProperties.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import org.springframework.stereotype.Component;
+import org.springframework.validation.annotation.Validated;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Getter
+@Setter
+@Component
+@Validated
```
Comment (sanjayparekh-hmcts) [inline src/main/java/uk/gov/hmcts/probate/config/properties/registries/RegistriesProperties.java]: and here

File: src/main/java/uk/gov/hmcts/probate/model/ApplicationType.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package uk.gov.hmcts.probate.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public enum ApplicationType {
```
Comment (sanjayparekh-hmcts) [inline src/main/java/uk/gov/hmcts/probate/model/ApplicationType.java]: is there some meaningful way to incorporate the PrintTemplateApplicationType enum functionality here too?

File: src/main/java/uk/gov/hmcts/probate/model/ccd/raw/request/CaseData.java [hunk 5]
```diff
@@ -215,4 +218,12 @@ private boolean isApplying(AdditionalExecutors ex, boolean applying) {
 
         return ex.getAdditionalExecutor().getAdditionalApplying().equals(applying ? YES : NO);
     }
+
+    public String getDeceasedFullName() {
```
Comment (sanjayparekh-hmcts) [inline src/main/java/uk/gov/hmcts/probate/model/ccd/raw/request/CaseData.java]: add these into CaseDataTest?

File: src/main/java/uk/gov/hmcts/probate/model/Event.java [hunk 6]
```diff
@@ -0,0 +1,6 @@
+package uk.gov.hmcts.probate.model;
+
+public enum Event {
```
Comment (sanjayparekh-hmcts) [inline src/main/java/uk/gov/hmcts/probate/model/Event.java]: is this enum really State, not Event? Also, maybe we need to refactor StateChangeService class for states to be included like these?

File: src/main/java/uk/gov/hmcts/probate/config/notifications/NotificationTemplates.java (hunk 1 above)
Comment (malinovsky) [reply src/main/java/uk/gov/hmcts/probate/config/notifications/NotificationTemplates.java]: https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-external-config.html#boot-features-external-config-validation

File: src/main/java/uk/gov/hmcts/probate/model/Event.java (hunk 6 above)
Comment (malinovsky) [reply src/main/java/uk/gov/hmcts/probate/model/Event.java]: It is Event.

**Pull request merged**